### DISCUSSION TITLE
Changing batch size of a text predictor

### DISCUSSION THREAD

**Discussion by marcmk6** [autogluon/autogluon#1582] [Q&A] [answered]
Hi,

I'm wondering if it's possible to change the batch size of a text predictor, just like setting `batch_size` in `fit` function of [ImagePredictor](https://auto.gluon.ai/stable/api/autogluon.task.html#autogluon.vision.ImagePredictor.fit) and [ObjectDetector](https://auto.gluon.ai/stable/api/autogluon.task.html#autogluon.vision.ObjectDetector.fit). I didn't find such information in documentation. 
It seems the default is 128 but it is too small for a large dataset and I got something like this:
```
2022-03-03 18:09:57,998 - root - Level 15 - [Iter 970/193900, Epoch 0] train loss=3.72e-01, gnorm=2.70e+00, lr=5.00e-06, #samples processed=124160, #sample per second=193.63. ETA=2125.58min
```
(I'm playing with ~3M training data on a V100)

Thanks

**Comment by Innixma** [comment, marked as answer]
We are refactoring TextPredictor in v0.4.0 release, so this response corresponses to v0.4.0 (or pre-release install) which will be available on March 9th.

Please refer to the customization tutorial which indicates how to specify the per-gpu batch size: https://auto.gluon.ai/dev/tutorials/text_prediction/customization.html#configure-textpredictor

**Reply by marcmk6** [reply]
Thanks for your work!

**Reply by marcmk6** [reply]
According to the [custom hyperparameters values section](https://auto.gluon.ai/dev/tutorials/text_prediction/customization.html#custom-hyperparameter-values) of the doc page you gave, I suppose I should use the `AutoMM` if I want to train a model **_from scratch_** with a batch size of, say, 1024. However `AutoMM` seems not in the current released version (installed from pip) as I got ModuleNotFound error. 
Will `AutoMM` also be available in the release in v0.4.0?

Thanks in advance!

**Reply by Innixma** [reply]
Yes, AutoMM will be available in v0.4.0
